single-node-refactor/src/Solvers/SGH_solver/src/contact.cpp

```diff
@@ -0,0 +1,463 @@
+#include "contact.h"
+
+// Definition of static member variables
+size_t contact_patch_t::num_nodes_in_patch;
+double contact_patches_t::bs;
+size_t contact_patches_t::n;
+
+KOKKOS_FUNCTION  // is called in macros
+void contact_patch_t::update_nodes(const mesh_t &mesh, const node_t &nodes, // NOLINT(*-make-member-function-const)
+                                   const corner_t &corner)
+{
+    // Constructing the points, vel_points, and internal_force arrays
+    for (int i = 0; i < 3; i++)
+    {
+        for (int j = 0; j < num_nodes_in_patch; j++)
+        {
+            const size_t node_gid = this->nodes_gid(j);
+
+            points(i, j) = nodes.coords(0, node_gid, i);
+            vel_points(i, j) = nodes.vel(0, node_gid, i);
+
+            // looping over the corners
+            for (size_t corner_lid = 0; corner_lid < mesh.num_corners_in_node(node_gid); corner_lid++)
+            {
+                size_t corner_gid = mesh.corners_in_node(node_gid, corner_lid);
+                internal_force(i, j) += corner.force(corner_gid, i);
+            }
+
+        }  // end local node loop
+    }  // end dimension loop
+}  // end update_nodes
+
+void contact_patch_t::capture_box(const double &vx_max, const double &vy_max, const double &vz_max,
+                                  const double &ax_max, const double &ay_max, const double &az_max,
+                                  const double &dt, CArrayKokkos<double> &bounds) const
+{
+    CArrayKokkos<double> add_sub(2, 3, contact_patch_t::num_nodes_in_patch);
+    FOR_ALL_CLASS(i, 0, contact_patch_t::num_nodes_in_patch, {
```

Comment:
Im not sure, but if the capture_box function if a KOKKOS_FUNCTION, then you will not be able to use the FOR_ALL_CLASS or any matar macro inside of it.  The assumption would be that everything inside of a KOKKOS_FUNCTION is serial.  If it isnt a KOKKOS_FUNCTION, and you knew this already, just tell me to shutup and go away. :)